### ISSUE TITLE
Segfaults with audio plugin

### ISSUE THREAD

**Issue by sergiobenrocha2** [libretro/RetroArch#2206]
If you select an audio filter, you get this backtrace:

```
#0  append_plugs (dsp=0x10357e0, dsp=0x10357e0, list=0x0, list=0x0) at audio/audio_dsp_filter.c:172
#1  rarch_dsp_filter_new (filter_config=filter_config@entry=0x7fffec56d1c1 "/usr/lib/retroarch/filters/audio/Chorus.dsp", sample_rate=29975) at audio/audio_dsp_filter.c:253
#2  0x00000000004260f1 in audio_driver_dsp_filter_init (device=device@entry=0x7fffec56d1c1 "/usr/lib/retroarch/filters/audio/Chorus.dsp") at audio/audio_driver.c:787
#3  0x0000000000418a7f in event_command (cmd=<optimized out>) at command_event.c:1308
#4  0x00000000004b3efc in general_write_handler (data=0xfcd7b0) at menu/menu_setting.c:2479
#5  0x00000000004b35cf in menu_setting_generic (setting=0xfcd7b0, wraparound=wraparound@entry=false) at menu/menu_setting.c:138
#6  0x00000000004b9913 in generic_action_ok (path=0x7fffffff5d60 "Chorus.dsp", id=<optimized out>, flush_id=<optimized out>, entry_idx=<optimized out>, idx=<optimized out>, 
    type=<optimized out>, label=<optimized out>) at menu/cbs/menu_cbs_ok.c:614
#7  0x0000000000496d66 in menu_entry_action (entry=entry@entry=0x7fffffff5d60, i=<optimized out>, action=action@entry=MENU_ACTION_OK) at menu/menu_entry.c:401
#8  0x00000000004cc0e8 in generic_menu_iterate (render_this_frame=<optimized out>, action=MENU_ACTION_OK) at menu/drivers/menu_generic.c:280
#9  0x000000000041c941 in rarch_main_iterate (sleep_ms=sleep_ms@entry=0x7fffffffdf20) at runloop.c:981
#10 0x0000000000412597 in rarch_main (argc=<optimized out>, argv=<optimized out>, data=0x0) at frontend/frontend.c:346
#11 0x00007ffff23ddec5 in __libc_start_main (main=0x40f080 <main>, argc=1, argv=0x7fffffffe048, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, 
    stack_end=0x7fffffffe038) at libc-start.c:287
#12 0x000000000040f0b0 in _start ()
```

```
Program received signal SIGSEGV, Segmentation fault.
append_plugs (dsp=0x10357e0, dsp=0x10357e0, list=0x0, list=0x0) at audio/audio_dsp_filter.c:172
172	   for (i = 0; i < list->size; i++)
(gdb) bt full
#0  append_plugs (dsp=0x10357e0, dsp=0x10357e0, list=0x0, list=0x0) at audio/audio_dsp_filter.c:172
        i = <optimized out>
        mask = <optimized out>
#1  rarch_dsp_filter_new (filter_config=filter_config@entry=0x7fffec56d1c1 "/usr/lib/retroarch/filters/audio/Chorus.dsp", sample_rate=29975) at audio/audio_dsp_filter.c:253
        dsp = 0x10357e0
#2  0x00000000004260f1 in audio_driver_dsp_filter_init (device=device@entry=0x7fffec56d1c1 "/usr/lib/retroarch/filters/audio/Chorus.dsp") at audio/audio_driver.c:787
No locals.
#3  0x0000000000418a7f in event_command (cmd=<optimized out>) at command_event.c:1308
        i = 0
        boolean = false
        driver = 0x7c1a60 <g_driver>
        global = 0x7963c0 <g_extern>
        settings = 0x7fffec565010
        system = <optimized out>
#4  0x00000000004b3efc in general_write_handler (data=0xfcd7b0) at menu/menu_setting.c:2479
        rarch_cmd = <optimized out>
        info = {list = 0x0, menu_list = 0x0, path = '\000' <repete 4095 vezes>, path_b = '\000' <repete 4095 vezes>, path_c = '\000' <repete 4095 vezes>, label = '\000' <repete 4095 vezes>, 
          exts = '\000' <repete 4095 vezes>, type = 0, type_default = 0, directory_ptr = 0, flags = 0, setting = 0x0}
        setting = 0xfcd7b0
        settings = <optimized out>
        driver = 0x7c1a60 <g_driver>
        global = <optimized out>
        menu_list = 0xc79780
        system = <optimized out>
        hash = <optimized out>
#5  0x00000000004b35cf in menu_setting_generic (setting=0xfcd7b0, wraparound=wraparound@entry=false) at menu/menu_setting.c:138
No locals.
#6  0x00000000004b9913 in generic_action_ok (path=0x7fffffff5d60 "Chorus.dsp", id=<optimized out>, flush_id=<optimized out>, entry_idx=<optimized out>, idx=<optimized out>, 
    type=<optimized out>, label=<optimized out>) at menu/cbs/menu_cbs_ok.c:614
        cbs = 0x1035380
        msg_force = false
        action_path = "/usr/lib/retroarch/filters/audio/Chorus.dsp", '\000' <repete 1773 vezes>...
        flush_type = 49
        ret = 0
        menu_path = 0x10bb320 "/usr/lib/retroarch/filters/audio/"
        flush_char = 0x0
        global = 0x7963c0 <g_extern>
        menu = 0xc83c70
        settings = 0x7fffec565010
        menu_list = 0xc79780
#7  0x0000000000496d66 in menu_entry_action (entry=entry@entry=0x7fffffff5d60, i=<optimized out>, action=action@entry=MENU_ACTION_OK) at menu/menu_entry.c:401
        ret = 0
        menu_list = 0xc79780
        cbs = <optimized out>
#8  0x00000000004cc0e8 in generic_menu_iterate (render_this_frame=<optimized out>, action=MENU_ACTION_OK) at menu/drivers/menu_generic.c:280
        selection = 1
        entry = {path = "Chorus.dsp\000sp\000in\000 (KHz)\000w", '\000' <repete 1255 vezes>..., 
          label = "\000\000\000\000\000\000\260=\000\000\000\000\000\000\260=\000\000\f?\000\000P>\000\200\021?\000\000P>\000\000\f?\000\000\060>\000\200\021?\000\000\060>\000\000\f?\000\000\060>\000\200\021?\000\000P>\000\000\026?\000\000\217>\000\200\033?\000\000\217>\000\000\026?\000\000\204>\000\200\033?\000\000\204>\000\000\026?\000\000\204>\000\200\033?\000\000\217>\000\000\000\000\000\000\260=\000\000\000\000\000\000\260=\000\000\000\000\000\000\260=\000\000\000\000\000\000\260=\000\000\000\000\000\000\260=\000\000\000\000\000\000\260=\000\000\360=\000\000P>\000\000\022>\000\000P>\000\000\360=\000\000\060>\000\000\022>\000\000\060>\000\000\360=\000\000\060>"..., value = '\000' <repete 4095 vezes>, entry_idx = 0, idx = 1, type = 11, spacing = 0}
        label = 0x10c6120 "audio_dsp_plugin"
        ret = 0
        label_hash = 0
        hash = <optimized out>
        menu = 0xc83c70
        menu_list = 0xc79780
#9  0x000000000041c941 in rarch_main_iterate (sleep_ms=sleep_ms@entry=0x7fffffffdf20) at runloop.c:981
        i = <optimized out>
        trigger_input = 256
        cmd = {fullscreen_toggle = false, overlay_next_pressed = false, grab_mouse_pressed = false, menu_pressed = false, quit_key_pressed = false, screenshot_pressed = false, 
          mute_pressed = false, osk_pressed = false, volume_up_pressed = false, volume_down_pressed = false, reset_pressed = false, disk_prev_pressed = false, disk_next_pressed = false, 
          disk_eject_pressed = false, movie_record = false, save_state_pressed = false, load_state_pressed = false, slowmotion_pressed = false, shader_next_pressed = false, 
          shader_prev_pressed = false, fastforward_pressed = false, hold_pressed = false, old_hold_pressed = false, state_slot_increase = false, state_slot_decrease = false, 
          pause_pressed = false, frameadvance_pressed = false, rewind_pressed = false, netplay_flip_pressed = false, cheat_index_plus_pressed = false, cheat_index_minus_pressed = false, 
          cheat_toggle_pressed = false}
        current = <optimized out>
        target = <optimized out>
        to_sleep_ms = <optimized out>
        last_input = 256
        driver = 0x7c1a60 <g_driver>
        settings = 0x7fffec565010
        input = 256
        system = 0x817860
        old_input = <optimized out>
#10 0x0000000000412597 in rarch_main (argc=<optimized out>, argv=<optimized out>, data=0x0) at frontend/frontend.c:346
        sleep_ms = 0
        args = 0x0
        ret = <optimized out>
        settings = 0x7fffec565010
        driver = <optimized out>
#11 0x00007ffff23ddec5 in __libc_start_main (main=0x40f080 <main>, argc=1, argv=0x7fffffffe048, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, 
    stack_end=0x7fffffffe038) at libc-start.c:287
        result = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {0, -8496816376815901526, 4255879, 140737488347200, 0, 0, 8496816377872164010, 8496844432632489130}, mask_was_saved = 0}}, priv = {pad = {
              0x0, 0x0, 0x5136c0 <__libc_csu_init>, 0x7fffffffe048}, data = {prev = 0x0, cleanup = 0x0, canceltype = 5322432}}}
        not_first_call = <optimized out>
#12 0x000000000040f0b0 in _start ()
No symbol table info available.
```

Is this related to #2202 ?

**Comment by sergiobenrocha2** [comment]
28b9ac7 broke it.

**Comment by sergiobenrocha2** [comment]
Fixed on 10b1d12
